src/main/reactjs/src/components/contactRequest/steps/WriteMessage.tsx

```diff
@@ -8,7 +8,7 @@ import {
 } from 'components/contactRequest/ContactRequestFormUtils';
```

Comment:
yksi vaihtoehto olisi laittaa handleMessageFieldErrors useEffectiin, silloin sitä tarkistetaan myös steppiä vaihtaessa

Reply:
Kokeilin nopeasti jotain tällaista:

```
import { useEffect, useRef, useState } from 'react';

import {
  ChosenTranslators,
  ChosenTranslatorsHeading,
  DisplayContactInfo,
  StepHeading,
} from 'components/contactRequest/ContactRequestFormUtils';
import { CustomTextField } from 'components/elements/CustomTextField';
import { H3 } from 'components/elements/Text';
import { translateOutsideComponent, useAppTranslation } from 'configs/i18n';
import { useAppDispatch, useAppSelector } from 'configs/redux';
import { TextFieldTypes } from 'enums/app';
import { ContactRequestFormStep } from 'enums/contactRequest';
import { useWindowProperties } from 'hooks/useWindowProperties';
import { setContactRequest } from 'redux/actions/contactRequest';
import { contactRequestSelector } from 'redux/selectors/contactRequest';
import { Utils } from 'utils';
import { StringUtils } from 'utils/string';

const getErrorForMessage = (message: string) => {
  const t = translateOutsideComponent();
  const error = Utils.inspectCustomTextFieldErrors(
    TextFieldTypes.Textarea,
    message,
    true
  );

  return error ? t(`akt.${error}`) : '';
};

export const WriteMessage = ({
  disableNext,
}: {
  disableNext: (disabled: boolean) => void;
}) => {
  // I18n
  const { t } = useAppTranslation({
    keyPrefix: 'akt.component.contactRequestForm',
  });
  const messageFieldRef = useRef<HTMLInputElement>();

  //Windows properties
  const { isPhone } = useWindowProperties();

  // State
  const [fieldError, setFieldError] = useState('');

  // Redux
  const { request } = useAppSelector(contactRequestSelector);
  const dispatch = useAppDispatch();

  useEffect(() => {
    const hasBlankMessage = StringUtils.isBlankString(request?.message);
    // Instead of relying on fieldError in local state, compute the error text
    // separately to ensure that we can disable the next button independently of
    // showing the error labels. The error label is to be shown only
    // after an onBlur event has occurred.
    const hasFieldError = getErrorForMessage(request?.message || '').length > 0;
    handleMessageFieldErrors();
    disableNext(hasBlankMessage || hasFieldError);
  }, [disableNext, fieldError, request]);

  const handleMessageFieldChange = () => {
    if (fieldError) {
      handleMessageFieldErrors();
    }

    dispatch(setContactRequest({ message: messageFieldRef.current?.value }));
  };

  const handleMessageFieldErrors = () => {
    if (messageFieldRef.current?.value)
      setFieldError(getErrorForMessage(messageFieldRef?.current?.value));
  };

  const getHelperMessage = () => {
    const value = request?.message;
    const errorToShow = fieldError ? `${fieldError}.` : '';
    const maxLength = Utils.getMaxTextAreaLength();

    return `${errorToShow} ${value?.length} / ${maxLength} ${t('characters')}`;
  };

  return (
    <div className="rows">
      <StepHeading step={ContactRequestFormStep.WriteMessage} />
      <div className="rows gapped">
        <ChosenTranslatorsHeading />
        <ChosenTranslators />
        {!isPhone && <DisplayContactInfo />}
        <div className="rows gapped">
          <H3>
            {t(
              `steps.${
                ContactRequestFormStep[ContactRequestFormStep.WriteMessage]
              }`
            )}
          </H3>
          <CustomTextField
            inputRef={messageFieldRef}
            id="contact-request-page__message-field"
            label={t('formLabels.writeMessageHere')}
            value={request?.message}
            type={TextFieldTypes.Textarea}
            onChange={handleMessageFieldChange}
            showHelperText
            helperText={getHelperMessage()}
            error={fieldError.length > 0}
            multiline
            fullWidth
            required
          />
        </div>
      </div>
    </div>
  );
};
```

Reply:
Joo, tämä ongelma oli kyllä tiedossa. Yksi vaihtoehto voisi tietty olla siirtää kenttien errorit reduxiin, jolloin ei ehkä tarvitsisi kikkailla `useRef` kanssa.

Reply:
jep, sekin on yksi vaihtoehto, mutta tässä kuitenkin useRef toimii aika hyvin?

Reply:
Ehkä sen saa toimimaan. Ihan täydellisesti ei vielä tuo versio toiminut. :)
![Screenshot 2022-04-06 at 9 25 28](https://user-images.githubusercontent.com/18038608/161909286-cd51e0f8-b600-4e4d-9b5b-a42548af9fbf.png)

Kokeilen vähän eri vaihtoehtoja.

Reply:
Mä muutin tuon nyt niin että virheviesti pidetään reduxissa `messageError` kentässä. Virheen pitäminen lokaalissa tilassa siten että se toimii konsistentisti onBlurin, onChangen ja stepperin kelauksen kanssa tuntui vaan liian monimutkaiselta.